### PR TITLE
 ExUI-623 Secure Document Store mode doesn't work properly during case creation(v2)

### PR DESCRIPTION
**Pull request by andywilkinshmcts**
### JIRA link (if applicable) ###
EXUI-623

### Change description ###
Jurisdiction Service needs to publish currently selected jurisdiction and case type as a BehaviorSubject

**Does this PR introduce a breaking change?** (check one with "x")

[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 1]
```diff
@@ -67,6 +73,10 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
 
   public ngOnInit(): void {
     this.secureModeOn = this.appConfig.getDocumentSecureMode();
+    console.info('writeDocumentField.ngInit: secure mode = ' + this.secureModeOn);
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:76]: @andywilkinshmcts - could probably remove this `console` statement

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 2]
```diff
@@ -188,10 +199,28 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+  // get the currently selected jurisdiction
   private subscribeToCaseDetails(): void {
-    this.caseEventSubscription = this.caseNotifier.caseView.subscribe({
-      next: (caseDetails) => {
-        this.caseDetails = caseDetails;
+    this.caseNotifierSubscription = this.caseNotifier.caseView.subscribe({
+      next: (caseDetails: CaseView) => {
+        this.caseTypeId = caseDetails?.case_id;
+        this.jurisdictionId = caseDetails?.case_type?.jurisdiction?.id;
+        console.info(`caseNotifier.next: caseType=${this.caseTypeId} juris=${this.jurisdictionId}`)
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:212]: @andywilkinshmcts - could probably remove this `console` statement

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 3]
```diff
@@ -188,10 +199,28 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+    });
+    this.jurisdictionSubs = this.jurisdictionService.selectedJurisdictionBS.subscribe({
+      next: (jurisdiction) => {
+        if (jurisdiction) {
+          this.jurisdictionId = jurisdiction.id;
+          if (jurisdiction.currentCaseType) {
+            this.caseTypeId = jurisdiction.currentCaseType.id
+          }
+        }
+        console.info(`selectedJurisdiction.next: caseType=${this.caseTypeId} juris=${this.jurisdictionId}`)
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:223]: @andywilkinshmcts - could probably remove this `console` statement

File: projects/ccd-case-ui-toolkit/src/lib/shared/services/jurisdiction/jurisdiction.service.ts [hunk 4]
```diff
@@ -19,7 +20,9 @@ export class JurisdictionService {
   }
 
   public announceSelectedJurisdiction(jurisdiction: Jurisdiction): void {
+    console.info ('Announcing selected jurisdiction = ' + jurisdiction?.id);
```
Comment (Daniel-Lam) [inline projects/ccd-case-ui-toolkit/src/lib/shared/services/jurisdiction/jurisdiction.service.ts:23]: @andywilkinshmcts - could probably remove this `console` statement

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@andywilkinshmcts - a few `console` statements that could be removed

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@andywilkinshmcts @RiteshHMCTS - Sonarcloud still reporting insufficient test coverage of 31.2%

**Review by Daniel-Lam — COMMENTED** [review]
@andywilkinshmcts - approved but with the caveat that the test coverage on new code is still too low at 17.3% (even though after merge the overall coverage will be 81.0%). Not great because merging this will bring the overall coverage down

**Review by Daniel-Lam — APPROVED** [review]

**Review by RiteshHMCTS — APPROVED** [review]

**Pull request merged**